gadget/layout.go

```diff
@@ -196,7 +213,19 @@ func LayoutVolumePartially(volume *Volume, constraints LayoutConstraints) (*Part
 
 // LayoutVolume attempts to completely lay out the volume, that is the
 // structures and their content, using provided constraints
-func LayoutVolume(gadgetRootDir string, volume *Volume, constraints LayoutConstraints) (*LaidOutVolume, error) {
+func LayoutVolume(gadgetRootDir, kernelRootDir string, volume *Volume, constraints LayoutConstraints) (*LaidOutVolume, error) {
+	var err error
+
+	var kernelInfo *kernel.Info
+	if !constraints.SkipResolveContent {
+		// Note that the kernelRootDir may reference the running
+		// kernel if there is a gadget update or the new kernel if
+		// there is a kernel update.
```

Comment:
this code path should fail now if kernelRootDir is ""  , this means changes the test to noResolveConstraints in some places

Reply:
I initially added a check here - but I suspect this PR is too early, the higher level manager and overlord tests fail. I'm investigating right now but I suspect it's just not enough "wiring up" at this level and this needs to be added to  #9875 - I update this comment ASAP. Sorry for the trouble

Reply:
I confirmed that this needs most of the update machinery in overlord before I can add the check. I will add a todo in #9875